src/pages/studyView/table/CNAGenesTable.tsx

```diff
@@ -1,106 +1,228 @@
+                <span className={classnames(data.alteration === -2 ? styles.del : styles.amp)}>
+                    {data.alteration === -2 ? 'DEL' : 'AMP'}
+                </span>,
+            sortBy: (data: CopyNumberCountByGene) => data.alteration,
+            defaultSortDirection: 'asc' as 'asc',
+            filter: (data: CopyNumberCountByGene, filterString: string, filterStringUpper: string) => {
+                return (data.alteration === -2 ? 'DEL' : 'AMP').indexOf(filterStringUpper) > -1;
+            },
+            width: 65
         },
         {
             name: '#',
             render: (data: CopyNumberCountByGene) =>
                 <LabeledCheckbox
                     checked={this.isChecked(data.entrezGeneId, data.alteration)}
-                    onChange={event => this.props.onUserSelection(data.entrezGeneId, data.alteration)}
+                    disabled={this.isDisabled(data.entrezGeneId, data.alteration)}
+                    onChange={event => this.togglePreSelectRow(data.entrezGeneId, data.alteration)}
                 >
                     {data.countByEntity}
                 </LabeledCheckbox>,
-            width: "20%"
+            sortBy: (data: CopyNumberCountByGene) => data.countByEntity,
+            defaultSortDirection: 'desc' as 'desc',
+            filter: (data: CopyNumberCountByGene, filterString: string, filterStringUpper: string) => {
+                return _.toString(data.countByEntity).indexOf(filterStringUpper) > -1;
+            },
+            width: 75
         },
         {
             name: 'Freq',
             render: (data: CopyNumberCountByGene) => <span>{data.frequency + '%'}</span>,
-            width: "20%"
+            sortBy: (data: CopyNumberCountByGene) => data.frequency,
+            defaultSortDirection: 'desc' as 'desc',
+            filter: (data: CopyNumberCountByGene, filterString: string, filterStringUpper: string) => {
+                return _.toString(data.frequency).indexOf(filterStringUpper) > -1;
+            },
+            width: 75
         }
     ];
 
+    @bind
     @action
     isChecked(entrezGeneId: number, alteration: number) {
-        var flag = false;
-        _.every(this.props.filters, (val: CopyNumberGeneFilterElement, index: number) => {
-            if (val.entrezGeneId === entrezGeneId && val.alteration === alteration) {
-                flag = true;
-                return false;
-            }
+        let record = _.find(this.preSelectedRows, (row: CNAGenesTableUserSelectionWithIndex) => row.entrezGeneId === entrezGeneId && row.alteration === alteration);
+        if (_.isUndefined(record)) {
+            return this.selectedRows.length > 0 && !_.isUndefined(_.find(this.selectedRows, (row: CNAGenesTableUserSelectionWithIndex) => row.entrezGeneId === entrezGeneId && row.alteration === alteration));
+        } else {
             return true;
-        });
-        return flag;
+        }
+    }
+
+    @bind
+    @action
```

Comment:
i don't see any mutation (setting/changing of state) here so does this need to be @action?